Add release notes for GOV.UK Frontend v3.12.0 (#2207)
alphagov/govuk-frontend: CHANGELOG.md

- want to avoid opting in if you have links within a [multiple-column layout](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Multiple-column_Layout) in your CSS - there is [a Chromium bug that affects such links](https://github.com/alphagov/govuk-frontend/issues/2204)

To enable this feature, set `$govuk-new-link-styles` to `true` before you import GOV.UK Frontend in your Sass:

```
// application.scss
$govuk-new-link-styles: true;
@import "govuk-frontend/all";
```

If you have created your own link styles, you should review them to ensure their use is consistent across your service.

To make it easier to be consistent in your use of link styles, we’ve also introduced new mixins and modifier classes for:

- removing underlines
- displaying white text on dark backgrounds

These new link styles will be enabled by default in a future major version of GOV.UK Frontend.

This was added in [pull request #2183: Updates to link styles and link hover states](https://github.com/alphagov/govuk-frontend/pull/2183).

### Style links to remove underlines

You can now remove underlines from links by using either the:

> - `govuk-link-style-no-underline` mixin in your Sass, or

Author: Need to link to Sass API reference content when that becomes available.

Reviewer: Should be http://frontend.design-system.service.gov.uk/sass-api-reference/#govuk-link-style-no-underline

> - `govuk-link--no-underline` class in your HTML

Author: Need to link class to Design System guidance when that becomes available.

Reviewer: Based on the current headings in https://github.com/alphagov/govuk-design-system/pull/1625, this should in theory be:

https://design-system.service.gov.uk/styles/typography/#links-without-underlines

An underline still appears when the user hovers their cursor over the link.

> This was added in [pull request #2183: Updates to link styles and link hover states](https://github.com/alphagov/govuk-frontend/pull/2183).

Reviewer: ```suggestion
This was added in [pull request #2214: Add no-underline mixin and modifier class](https://github.com/alphagov/govuk-frontend/pull/2214).
```

### Style links on dark backgrounds

You can now style links on dark backgrounds to use white text colour by using either the:

> - `govuk-link-style-inverse` mixin in your Sass, or

Author: Need to link to Sass API reference content when that becomes available.

Reviewer: http://frontend.design-system.service.gov.uk/sass-api-reference/#govuk-link-style-inverse

> - `govuk-link--inverse` class in your HTML

Author: Need to link class to Design System guidance when that becomes available.

Reviewer: Based on the current headings in https://github.com/alphagov/govuk-design-system/pull/1625, this should in theory be:

https://design-system.service.gov.uk/styles/typography/#links-on-dark-backgrounds

> - `govuk-link--inverse` class in your HTML

Reviewer: ```suggestion
- `govuk-link--inverse` class in your HTML

This was added in [pull request #2213: Add inverse link mixin and modifier class](https://github.com/alphagov/govuk-frontend/pull/2213).
```

- want to avoid opting in if you have links within a [multiple-column layout](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Multiple-column_Layout) in your CSS - there is [a Chromium bug that affects such links](https://github.com/alphagov/govuk-frontend/issues/2204)

To enable this feature, set `$govuk-new-link-styles` to `true` before you import GOV.UK Frontend in your Sass:

```
// application.scss
$govuk-new-link-styles: true;
@import "govuk-frontend/all";
```

If you have created your own link styles, you should review them to ensure their use is consistent across your service.

To make it easier to be consistent in your use of link styles, we’ve also introduced new mixins and modifier classes for:

- removing underlines
- displaying white text on dark backgrounds

These new link styles will be enabled by default in a future major version of GOV.UK Frontend.

This was added in [pull request #2183: Updates to link styles and link hover states](https://github.com/alphagov/govuk-frontend/pull/2183).

#### Style links to remove underlines

You can now remove underlines from links by using either the:

> - `govuk-link-style-no-underline` mixin in your Sass, or

Author: ```suggestion
- [`govuk-link-style-no-underline` mixin](http://frontend.design-system.service.gov.uk/sass-api-reference/#govuk-link-style-no-underline) in your Sass, or
```

> - `govuk-link-style-no-underline` mixin in your Sass, or

Author: ```suggestion
- [`govuk-link-style-no-underline` mixin](http://frontend.design-system.service.gov.uk/sass-api-reference/#govuk-link-style-no-underline) in your Sass, or
```

> - `govuk-link--no-underline` class in your HTML

Author: ```suggestion
- [`govuk-link--no-underline` class](https://design-system.service.gov.uk/styles/typography/#links-without-underlines) in your HTML
```

An underline still appears when the user hovers their cursor over the link.

This was added in [pull request #2214: Add no-underline mixin and modifier class](https://github.com/alphagov/govuk-frontend/pull/2214).

#### Style links on dark backgrounds

You can now style links on dark backgrounds to use white text colour by using either the:

> - `govuk-link-style-inverse` mixin in your Sass, or

Author: ```suggestion
- [`govuk-link-style-inverse` mixin](http://frontend.design-system.service.gov.uk/sass-api-reference/#govuk-link-style-inverse) in your Sass, or
```

> - `govuk-link--inverse` class in your HTML

Author: ```suggestion
- [`govuk-link--inverse` class](https://design-system.service.gov.uk/styles/typography/#links-on-dark-backgrounds) in your HTML
```
